android/src/com/mapswithme/maps/location/SensorHelper.java

```diff
@@ -39,46 +37,54 @@ public void onSensorChanged(SensorEvent event)
     if (!MwmApplication.get().arePlatformAndCoreInitialized())
       return;
 
-    if (mCompassParams.isDataAbsent())
-      notifyCompassImmediately(event);
+    if (mSensorData.isAbsent())
+    {
+      notifyImmediately(event);
+      return;
+    }
+
```

Comment:
AFAIK you intended to skip all SensorType.TYPE_MAGNETIC_FIELD SensorEvents for a period of 10ms?
So handle one and push the "RateLimitMessage" into the Queue after you actually _handled_ it please.

`    if (mSensorData.isAbsent())
    {
      notifyImmediately(event);
    }
    else if (SensorType.get(event) == SensorType.TYPE_ACCELEROMETER)
    {
      notifyImmediately(event);
    }
    else if (!mHandler.hasMessages(MARKER))
    {
      notifyImmediately(event);
      addRateLimitMessage();
    }`